### DISCUSSION TITLE
Services, pods and VCN cidrs

### DISCUSSION THREAD

**Discussion by codex70** [oracle-terraform-modules/terraform-oci-oke#613] [Q&A]
It seems that we're not able to create a cluster without a class A subnet for each of the services, pods and the VCN. The problem with this, in an internal network, is that we will quickly use up a lot of the available class A subnets.

I can easily foresee a situation whereby we would create separate clusters for different clients to totally separate their deployments. This would be completely prohibitive with 3 entire class A subnets per cluster.

Is there something that I'm missing here, is there any way to deploy into a smaller space, or is this really the preferred approach?

**Comment by hyder** [comment]
Apologies for the late answer. I've successfully created a cluster with class B CIDR range for the VCN. I didn't change the services and pods though. 

Can you please share the ranges you used?

**Comment by codex70** [comment]
Hi Ali, thanks for getting back to me. The configuration I was using is as follows:

```
vcn_cidrs       = ["10.12.0.0/16"]
pods_cidr      = "10.13.0.0/16"
services_cidr  = "10.14.0.0/16"
```
If I remember rightly the pods and services complain if you try to assign any subnet range smaller than a `/16` (so `/17` or higher). It should be easy to test.
